### ISSUE TITLE
Possible Bug: response is not defined after the first http request

### ISSUE THREAD

**Issue by woolfas** [AnWeber/httpyac#388]
Steps to reproduce:
1. Create http_scripts/response.http script:

```
POST https://httpbin.org/anything
Content-Type: application/json

{
        "id": "0001",
        "type": "donut",
        "name": "Cake"
}

{{ $global['responseData'] = response.parsedBody }}
```
2. Execute it in Cli: `httpyac -a "http_scripts/response.http"`

Expected results: `$global['responseData']` has `response.parsedBody` assigned
Actual results: Httpyac throws exception `"ReferenceError: response is not defined"`

However this script is executed without exceptions:

```
POST https://httpbin.org/anything
Content-Type: application/json

{
        "id": "0001",
        "type": "fruit",
        "name": "Apple"
}

POST https://httpbin.org/anything
Content-Type: application/json

{
        "id": "0001",
        "type": "donut",
        "name": "Cake"
}

{{ $global['responseData'] = response.parsedBody }}
```
As a result, `$global['responseData']` contains response of the first HTTP request but not the second as expected. So it looks like that 'response' object is created only after the script is executed but not before.

And here's the last attempt which actually worked well for me:
```
POST https://httpbin.org/anything
Content-Type: application/json

{
        "id": "0001",
        "type": "donut",
        "name": "Cake"
}

{{
    $global['responseData'] = response.parsedBody
}}
```
So the difference here is that the script opening and closing brackets `{{ ... }}` are place on a separate lines. In the previous examples script was written within the same line as the brackets.

So it looks that sometimes it is complicated to understand at which point the script, which is written after request line, is executed - before or after actual HTTP request. That can lead to unwanted results. In the first 2 examples it seems that script is treated as a part of request body generation (which is actually a good feature and I use it a lot) and executed before actual HTTP request happens. And in the last example script is treated as a response processing script. So the behavior depends only on where you will put the brackets.

However if this is expected feature, then it would be nice to have that described in the documentation. Also maybe it would be good to have a possibility to explicitly define where request generation script ends and where response processing script starts.

**Comment by AnWeber** [comment]
@woolfas The distinction for me is really the line break after `{{` and the script content. If there is no line break, you want to use templating mechanism (and is executed before request). This feature is needed to allow some simplification for response body

```
{{
  exports.body= {
   foo: "bar"
  }
}}
POST <some_url>
{{body}}
```

I don't see the need for an explicit separation between response body and script area. This would be visual only, as Intellij Http Client does not provide this either and I want to remain compatible. You could specially mark the script area, but it would change the execution time (executed before response display).

```
POST https://httpbin.org/anything

{
        "id": "0001",
        "type": "donut",
        "name": "Cake"
}

{{@response
  console.info("on every response");
}}
```

**Comment by AnWeber** [comment]
Maybe as a tip. In the outline editor I show the distinction between request body and script area
<img width="227" alt="image" src="https://user-images.githubusercontent.com/13434739/216786646-a748ee5c-5cc5-4d79-8e73-11adfd8d44ce.png">

```
POST  https://httpbin.org/json

{
  "test": 1
}

{{
  // post request script
  exports.header = response.headers['content-type'];
}}
```
